library/src/scala/quoted/Quotes.scala

```diff
@@ -21,7 +22,23 @@ transparent inline def quotes(using q: Quotes): q.type = q
  *
  *  It contains the low-level Typed AST API metaprogramming API.
  *  This API does not have the static type guarantees that `Expr` and `Type` provide.
+ *
+ *  `Quotes` are generated from an enclosing `${ ... }` or `scala.staging.run`. For example:
+ *  ```scala sc:nocompile
+ *  import scala.quoted._
+ *  inline def myMacro: Expr[T] =
+ *    ${ /* (quotes: Quotes) ?=> */ myExpr }
+ *  def myExpr(using Quotes): Expr[T] =
+ *    '{ f(${ /* (quotes: Quotes) ?=> */ myOtherExpr }) }
+ *  }
+ *  def myOtherExpr(using Quotes): Expr[U] = '{ ... }
+ *  ```
  */
+@implicitNotFound("""No given instance of type scala.quoted.Quotes was found.
+
+Quotes are generated from an enclosing splice `$ { ... }` or `scala.staging.run` call.
+Maybe that splice `$ { ... }` or `scala.staging.run` is missing?
```

Comment:
I had to use `$ {` because I could not find a way to write `${` within the message of `implicitNotFound`. It assumes that `${` is the start of a type interpolation. Escape `\`,`$` characters or escape Unicode do not work because the string is interpreted raw.